### PR TITLE
feat: Add cache toggle to MCPToolsComponent and tests

### PR DESCRIPTION
**Pull request by edwinjosechittilappilly**
Introduces a 'use_cache' BoolInput to MCPToolsComponent, allowing users to enable or disable caching of MCP tools for performance or freshness. Updates the component logic to respect the cache toggle and adds comprehensive unit tests covering cache enabled, disabled, toggle, and edge cases.

### CODE REVIEW

**Comment by edwinjosechittilappilly** [comment]
@coderabbitai review the pr

File: src/lfx/src/lfx/components/agents/mcp_component.py [hunk 1]
```diff
@@ -68,6 +69,13 @@ def _ensure_cache_structure(self):
             info="Select the MCP Server that will be used by this component",
             real_time_refresh=True,
         ),
+        BoolInput(
+            name="use_cache",
+            display_name="Use Cache",
```
Comment (jordanrfrazier) [inline src/lfx/src/lfx/components/agents/mcp_component.py]: Discussing naming and default behavior

File: src/lfx/src/lfx/components/agents/mcp_component.py [hunk 2]
```diff
@@ -132,16 +143,31 @@ async def update_tool_list(self, mcp_server_value=None):
             self.tools = []
             return [], {"name": server_name, "config": server_config_from_value}
 
-        # Use shared cache if available
-        servers_cache = safe_cache_get(self._shared_component_cache, "servers", {})
-        cached = servers_cache.get(server_name) if isinstance(servers_cache, dict) else None
+        # Check if caching is enabled
+        use_cache = getattr(self, "use_cache", True)
```
Comment (jordanrfrazier) [inline src/lfx/src/lfx/components/agents/mcp_component.py]: If, for whatever reason, `use_cache` is not set, this will default to `True`, whereas in the Input we default to `False`. We should set this to `False` just to avoid confusion later on.

File: src/lfx/src/lfx/components/agents/mcp_component.py [hunk 3]
```diff
@@ -132,16 +143,31 @@ async def update_tool_list(self, mcp_server_value=None):
...
-            server_config_from_value = cached["config"]
-            return self.tools, {"name": server_name, "config": server_config_from_value}
+            try:
+                self.tools = cached["tools"]
+                self.tool_names = cached["tool_names"]
+                self._tool_cache = cached["tool_cache"]
+                server_config_from_value = cached["config"]
+            except (TypeError, KeyError, AttributeError) as e:
+                # Handle corrupted cache data by clearing it and continuing to fetch fresh tools
+                await logger.awarning(f"Corrupted cache data for server {server_name}: {e}")
```
Comment (jordanrfrazier) [inline src/lfx/src/lfx/components/agents/mcp_component.py]: "Corrupted" is a bit of a scary term. It'd probably change this to something like "Unable to use cached data for MCP Server, fetching updated ..."

File: src/lfx/src/lfx/components/agents/mcp_component.py [hunk 4]
```diff
@@ -294,14 +322,21 @@ async def update_build_config(self, build_config: dict, field_value: str, field_
...
+                            cached = servers_cache.get(current_server_name)
+                            if cached is not None:
+                                try:
+                                    cached_tools = cached["tools"]
+                                    self.tools = cached_tools
+                                    self.tool_names = cached["tool_names"]
+                                    self._tool_cache = cached["tool_cache"]
+                                except (TypeError, KeyError, AttributeError) as e:
+                                    # Handle corrupted cache data by ignoring it
+                                    await logger.awarning(f"Corrupted cache data for server {current_server_name}: {e}")
```
Comment (jordanrfrazier) [inline src/lfx/src/lfx/components/agents/mcp_component.py]: same
